kafka-connect-s3/docs/configuration_options.rst

```diff
@@ -109,6 +109,14 @@ S3
   * Valid Values: Any class implementing: interface com.amazonaws.auth.AWSCredentialsProvider
   * Importance: low
 
+``s3.credentials.provider.configs.``
```

Comment:
The only one reason I've added `config` to the prefix - it allows to isolate properties easily needed by a specific credential provider from a value of `s3.credentials.provider.class` property without the extra code.
Otherwise, if a user has configs:
* `s3.credentials.provider.class=com.acme.SampleProvider`
* `s3.credentials.provider.key1=value`
* `s3.credentials.provider.key2=value`,
* `other.important.configs.foo=bar`
* ...
 and we will filter them out by `s3.credentials.provider.` - we would get 
* `s3.credentials.provider.class=com.acme.SampleProvider`
* `s3.credentials.provider.key1=value`
* `s3.credentials.provider.key2=value` 
passed down to `com.acme.SampleProvider`. It seems redundant to me to pass `s3.credentials.provider.class` property.

We can also remove `s3.credentials.provider.class` from a filtered set before calling `configure()`, but it seems
 not very elegant.

Any thoughts?

Reply:
The pattern we use in quite a few places is to not have `.class` in the property that defines the fully qualified class name. Examples include `value.converter`, `key.converter`, `header.converter`, and so on. Then properties for that implementation start with the same prefix (with an additional `.`), so there's nothing to remove. I personally like this option.

However, the S3/HDFS connector do use the `.class` suffix on the property that defines the class. On one hand, it's nice that it's clear that the value should be a class name, but on the other it makes this a bit more clunky.

Given this, I'd suggest that we follow the existing S3 convention and use the `.class` suffix, but remove the property that defines the class before passing to the implementation's `configure` method. Yes, it's clunky, but it follows the conventions and makes for a better user experience IMO.

Reply:
I agree with @rhauch and I believe we should take the - admittedly small - burden ourselves, in terms of util methods or code that strips the `.class` suffix if present in a `Type.CLASS` property and forwards all the properties with that extracted prefix besides the class to the `configure` method. The alternative, which is forcing everyone to use a redundant element in the prefix feels unnecessary here.

Reply:
Sure, that makes sense from user perspective, so will implement suggested approach.